execution/group_by.go

```diff
@@ -280,7 +302,11 @@ func (gb *GroupByStream) Trigger(ctx context.Context, tx storage.StateTransactio
 
 	// Set IDs for records
 	for i := range output {
-		WithID(NewRecordIDFromStreamIDWithOffset(gb.streamID, triggeredCountValue+i))(output[i])
+		keyHash, err := key.Hash()
```

Comment:
Could you explain what was wrong with the previous ID setting?

Reply:
record id's were duplicated, because every storage in processFunction is prefixed with event time, so `triggeredCount` storage also is which means that for given records with event times
`1 1 1 2 2 3 3`
instead of record ids
`0 1 2 3 4 5 6`
you receive
`0 1 2 0 1 0 1`
that's why there is key hass added to record id to prevent duplications